### PR TITLE
logback graceful shutdown

### LINKED ISSUE #8277: Async appender should use shutdown hook for logback graceful shutdown
This applies to all Play 2.x versions:

The async appender in `logback.xml` uses an async appender:

https://github.com/playframework/playframework/blob/master/framework/src/play-logback/src/main/resources/logback-play-default.xml#L15

```xml
  <appender name="ASYNCSTDOUT" class="ch.qos.logback.classic.AsyncAppender">
    <appender-ref ref="STDOUT" />
  </appender>
```

There's a note saying that when logback closes down, there's a small chance that logged messages will still be in the queue, because Logback does not do graceful shutdown by default.  

The logstash-logback-encoder notes reference this "In order to guarantee that logged messages have had a chance to be processed by asynchronous appenders (including the TCP appender) and ensure background threads have been stopped, you'll need to [cleanly shut down logback](https://logback.qos.ch/manual/configuration.html#stopContext) when your application exits." https://github.com/logstash/logstash-logback-encoder/tree/logstash-logback-encoder-4.11#async-appenders

Cleanly in this case means adding a shutdown hook https://logback.qos.ch/manual/configuration.html#shutdownHook but you must specify `DelayingShutdownHook` explicitly, otherwise you'll get a warning https://jira.qos.ch/browse/LOGBACK-1090

```xml
     <shutdownHook class="ch.qos.logback.core.hook.DelayingShutdownHook"/>
```

### PR DESCRIPTION
**Pull request by YuitoSato**
# Pull Request Checklist

* [x] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
* [x] Have you read through the [contributor guidelines](https://www.playframework.com/contributing)?
* [x] Have you signed the [Lightbend CLA](https://www.lightbend.com/contribute/cla)?
* [x] Have you referenced any issues you're fixing using [commit message keywords](https://help.github.com/articles/closing-issues-using-keywords/)?
* [x] Have you added copyright headers to new files?
* [x] Have you checked that both Scala and Java APIs are updated?
* [x] Have you updated the documentation for both Scala and Java sections?
* [x] Have you added tests for any changed functionality?

# Helpful things

## Fixes

Fixes #8277

## Purpose

This PR adds graceful shutdown setting to the default logback which uses AsyncAppender.

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
Hi @YuitoSato,

Thank you for your contribution! We really value the time you've taken to put this together.

Before we proceed with reviewing this pull request, please sign the Lightbend Contributors License Agreement:

[http://www.lightbend.com/contribute/cla](http://www.lightbend.com/contribute/cla)

**Comment by YuitoSato** [comment]
done with signing the CLA.

File: documentation/manual/working/commonGuide/configuration/SettingsLogger.md [hunk 1]
```diff
@@ -18,6 +18,7 @@ A few things to note about these configurations:
 * These default configs specify only a console logger which outputs only 10 lines of an exception stack trace.
 * Play uses ANSI color codes by default in level messages.
 * For production, the default config puts the console logger behind the logback [AsyncAppender](https://logback.qos.ch/manual/appenders.html#AsyncAppender).  For details on the performance implications on this, see this [blog post](https://blog.takipi.com/how-to-instantly-improve-your-java-logging-with-7-logback-tweaks/).
+* In order to guarantee that logged messages have had a chance to be processed by asynchronous appenders (including the TCP appender) and ensure background threads have been stopped, you'll need to cleanly shut down logback when your application exits.
```
Comment (wsargent) [inline documentation/manual/working/commonGuide/configuration/SettingsLogger.md]: This is good, but we want to have the line explicitly included:

```
<shutdownHook class="ch.qos.logback.core.hook.DelayingShutdownHook"/>
```

along with a link to https://logback.qos.ch/manual/configuration.html#shutdownHook and https://jira.qos.ch/browse/LOGBACK-1090

**Comment by YuitoSato** [comment]
Thanks @wsargent. You mean that it is necessary to include the two links, don't you? Am I supposed to explain why DelayingShutdownHook should be used in the document?

**Comment by marcospereira** [comment]
2.6.x: 3aa49305ec7237c7e1d8e39bb0b88308faa3b747

**Comment by mkurz** [comment]
Starting with Play 2.7, adding `<shutdownHook />` to the logback config is very bad idea, because you very likely even miss log entries. As of 2.7, Play handles the shutdown of the logger context with it's coordinated shutdown hooks. See #11532 where I revert his pull request (and more explanation).
(Of course back in the day when this PR was merged it might was a good idea)

**Pull request merged**